wayland/getting-started.txt

```diff
@@ -0,0 +1,249 @@
+| Qt             | QT_QPA_PLATFORM=xcb,wayland                                 |
++----------------+-------------------------------------------------------------+
+
+
+[2.0] Xorgless
+________________________________________________________________________________
+
+Ensure no Xorg packages are installed, and remove any Xorg repositories from
+$KISS_PATH, keeping only xkeyboard-config and libxkbcommon. You will have to
+modify the depends files of many packages (including mesa, freetype-harfbuzz,
+pango, cairo, etc.) to not include any Xorg dependencies, but few modifications
+are required beyond that.
+
+First, build wayland and wayland-protocols.
+
+Then, build xkeyboard-config, followed by libxkbcommon with wayland support and
+without x11 support.
+
+Finally, build mesa with the wayland platform and without the x11 platform.
+
+Some packages might assume that, because you are building Linux, you must be
+using GLX. GLX is a graphics implementation provided by mesa, but is only built
+on Xorg systems. Wayland uses EGL and GLES. For some packages, like cairo, you
+might have to specify the gl-backend to be used. 
+
+Launching applications in Wayland requires a little more intervention. GTK will
+default to Wayland usually, but adding the following to your environment is a
+good idea
+
++----------------+-------------------------------------------------------------+
+| GTK            | GDK_BACKEND=wayland                                         |
+| Qt             | QT_QPA_PLATFORM=wayland                                     |
+| firefox        | MOZ_ENABLE_WAYLAND=wayland                                  |
++----------------+-------------------------------------------------------------+
+
+
+As far as finding alternatives to traditionally popular Xorg software like dmenu
+or st, checkout lists like [9] and [16]. 
+
+
+[3.0] Post installation
+________________________________________________________________________________
+
+In environments without seatd/systemd/elogind/consolekit the XDG_RUNTIME_DIR
+variable must be set manually. This should be set before trying to launch a
+wayland session. Add something like this to your user's $HOME/.profile:
+
++------------------------------------------------------------------------------+
+|                                                                              |
+|   mkdir -pm 0700 "${XDG_RUNTIME_DIR:=/tmp/$(id -u)-runtime-dir               |
+|   export XDG_RUNTIME_DIR                                                     |
+|                                                                              |
++------------------------------------------------------------------------------+
+
+Install a compositor. Below is a list:
+
++----------+-------------------------------------------------------------------+
+| dwl      |  A dwm-like compositor using wlroots [17]                         |
+| river    |  A compositor inspired by dwm and bspwm using wlroots [18]        |
```

Comment:
I *think* as is the zig language cannot be build yet on KISS. There was some work by @mmatango but is not yet been resolved?
https://github.com/kisslinux/community/issues/1623

Reply:
`Zig` cannot build on `KISS` yet, it requires some changes to `LLVM`,
`CLANG` and `LLD` to make it compile and link properly.

However the `zig` team did pick up my issue and it was upstreamed to be worked on in time for the `0.8.0` prerelease.

Reply:
You've discovered that I've been secretly hoping @mmatongo's issue would get fixed soon :v

Reply:
By July it should be fixed, I could try and compile and package it as a bin.

Reply:
Does the issue exist beyond it (currently?) requiring all targets to be built? Your issue here https://github.com/ziglang/zig/issues/7209 gives me the impression zig expects static libLLVM. I thought I remembered you making an issue about this with Dylan but I can't seem to find it. 

We should also probably move this type of discussion to https://github.com/kiss-community/repo-main